src/main/java/bsh/security/MainSecurityGuard.java

```diff
@@ -0,0 +1,224 @@
+package bsh.security;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import bsh.Interpreter;
+
+/** It prevents the execution of codes that manipulate the SecurityGuard or MainSecurityGuard */
+class BasicSecurityGuard implements SecurityGuard {
```

Comment:
Could these not be default methods on the interface, to be overwritten by custom security guard?

Reply:
In other words, if the methods of `BasicSecurityGuard` are implemented as default methods on interface `SecurityGuard` then we can remove the need for the `BasicSecurityGuard` class.

Reply:
I guess no, the `SecurityGuard` is just a generic interface, the validation logic to block the execution of something still must be implemented, the `BasicSecurityGuard` is an internal logic validation to prevent that code interpreted by BeanShell can interact with the API of Security ( add or remove a `SecurityGuard` ).

But I can put a default implementation to `SecurityGuard` methods to avoid this boilerplate of `return true;` at so many methods.

Reply:
Ok I see what you mean